packages/babel-plugin-transform-async-to-generator/test/fixtures/async-to-generator/deeply-nested-asyncs/output.js

```diff
@@ -1,38 +1,31 @@
+var _s;
 function s(_x) {
-  return _s.apply(this, arguments);
-}
-function _s() {
-  _s = babelHelpers.asyncToGenerator(function* (x) {
+  return (_s = _s || babelHelpers.asyncToGenerator(function* (x) {
     var _arguments = arguments,
-      _this = this;
+      _this = this,
+      _ref;
     for (var _len = arguments.length, args = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
       args[_key - 1] = arguments[_key];
     }
-    let t = /*#__PURE__*/function () {
```

Comment:
It should still be a pure function, we may need to update the annotation logic here.

Reply:
In this case `t` is a function and not a function call, so tools can already infer that it's pure (because all function expressions are pure).

Example from terser, removing the usages of `t`:
```js
var _s;function s(r){return(_s=_s||babelHelpers.asyncToGenerator((function*(r){for(var n=arguments.length,e=new Array(n>1?n-1:0),s=1;s<n;s++)e[s-1]=arguments[s];return this.h()}))).apply(this,arguments)}
```